corehq/apps/domain/forms.py

```diff
@@ -1331,7 +1331,22 @@ class HQSetPasswordForm(SetPasswordForm):
                                     """))
 
     def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+        from corehq.apps.hqwebapp.utils import decode_password
+        password1 = decode_password(self.cleaned_data.get('new_password1'))
+        return clean_password(password1)
+
+    def clean_new_password2(self):
```

Comment:
Looks like this function is duplicated in about 3 places. Would it makes sense to refactor to remove the duplication?